src/graphHasCycles.ts

```diff
@@ -0,0 +1,38 @@
+import { PGraphNodeWithDependencies } from "./types";
+
+/**
+ * Checks for any cycles in the dependency graph, returning false if no cycles were detected.
+ */
+export function graphHasCycles(pGraphDependencyMap: Map<string, PGraphNodeWithDependencies>, nodesWithNoDependencies: string[]): boolean {
+  const checkForCyclesInternal = (currentNodeId: string, visitedNodes: Set<string>): boolean => {
```

Comment:
Wondering if you can make this an iterative algo so it's a bit more scalable?

Reply:
Done, had to write an iterative toposort for the priority computation. Cycle detection was trivial.